External/Traffic/Source/MassTraffic/Private/MassTrafficMovement.cpp

```diff
@@ -385,6 +385,108 @@ float TimeToCollision(
 	return (Tau < 0) ? TNumericLimits<float>::Max() : Tau;
 }
 
+float TimeToCollisionBox2D(
+	const FTransform& AgentTransform, const FVector& AgentVelocity, const FBox2D& AgentBox,
+	const FTransform& ObstacleTransform, const FVector& ObstacleVelocity, const FBox2D& ObstacleBox)
+{
+	const FBox AgentWorldBox = FBox(FVector(AgentBox.Min, 0.0), FVector(AgentBox.Max, 0.0)).TransformBy(AgentTransform);
+	const FBox ObstacleWorldBox = FBox(FVector(ObstacleBox.Min, 0.0), FVector(ObstacleBox.Max, 0.0)).TransformBy(AgentTransform);
+	if (AgentWorldBox.IntersectXY(ObstacleWorldBox))
```

Comment:
What about overlap?

Reply:
The IntersectXY function includes the case where one includes the other entirely.